app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,46 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
+    @user = current_user
     @posts = @user.posts.order(created_at: :asc)
+    @current_user = current_user
   end
 
   def show
     @post = Post.find(params[:id])
+    @user = @post.author
     @comments = @post.recent_comments
+    @comment = Comment.new
+    @current_user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @user = @post.author
+    @like = Like.new(author_id: @user.id, post_id: @post.id)
+    @like.save
+    redirect_to user_post_path(user_id: @user.id, id: @post.id)
+  end
+
+  def create
+    puts current_user.inspect
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.comments_counter = 0
+    @post.likes_counter = 0
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post created successfully.'
+    else
+      render :new
+    end
+  end
```

Comment:
- [ ] I notice that every time I create a post for the second user which is `Lily` on my list the post goes to the first user, I tried to create other users and when I create a post for other users the post goes to the first user `Tom`. Please kindly fix that 🙏🏾 

https://github.com/SaneMyburg/blog-app/assets/74501523/9affaf94-76ac-4e54-a882-e5e023dcb1f2


https://github.com/SaneMyburg/blog-app/assets/74501523/34a316a7-7bbc-49ab-8cce-a438b3c7f608

Reply:
@cynthiainga, thanks for the review I appreciate your time taken, however, it is my understanding that the logged-in user, is the first user who is Tom, so Lilly can't create a post.